crates/ruff_python_formatter/src/comments/mod.rs

```diff
@@ -324,12 +324,21 @@ impl<'a> Comments<'a> {
         let map = if comment_ranges.is_empty() {
             CommentsMap::new()
         } else {
-            CommentsVisitor::new(source_code, comment_ranges).visit(root)
+            get_comments_map(root, source_code, comment_ranges)
         };
 
         Self::new(map)
     }
 
+    /// Extracts the comments from the AST.
+    pub(crate) fn collect_decorated_comments(
+        root: &'a Mod,
+        source_code: SourceCode<'a>,
+        comment_ranges: &'a CommentRanges,
+    ) -> Vec<DecoratedComment<'a>> {
+        collect_comments(root, source_code, comment_ranges)
+    }
+
```

Comment:
This method feels misplaced because it doesn't involve `Comments` at all. Why not expose `collect_comments` directly?